lang/mosml/files/0004-Address-lvalue-error.patch

```diff
@@ -0,0 +1,22 @@
+From 6a910c2115c9a9722a64baee9f2f427d804685c9 Mon Sep 17 00:00:00 2001
+From: barracuda156 <vital.had@gmail.com>
+Date: Wed, 26 Jul 2023 14:39:16 +0800
+Subject: [PATCH] Address lvalue error
+
+
+diff --git src/dynlibs/mmysql/mmysql.c src/dynlibs/mmysql/mmysql.c
+index 083771c..621c6ab 100644
+--- src/dynlibs/mmysql/mmysql.c
++++ src/dynlibs/mmysql/mmysql.c
+@@ -76,9 +76,9 @@ void dbresult_finalize(value dbresval)
+   MYSQL_ROW_OFFSET* index = DBresultindex_val(dbresval);
+   if (dbres != NULL) {
+     mysql_free_result(dbres);
+-    DBresult_val(dbresval) = NULL;
++    DBresult_val(dbresval) == NULL;
+     stat_free((char*)index);
+-    DBresultindex_val(dbresval) = NULL;
++    DBresultindex_val(dbresval) == NULL;
```

Comment:
this is not a functionally equivalent change.

Reply:
What will be a correct fix? Existing code fails for years.

Reply:
(I know it is not equivalent, but I am not sure if the original code is correct to begin with.)

Reply:
That's good that you see it is not equivalent. So this is not a patch that can be used as it is. Code changes that make things compile are not necessarily correct code changes -- an obvious statement, but every time you make a patch you have to look at it and see what you've done. Is it right?

I would take this issue upstream and see how they want to handle the issue. If there is no more upstream, then someone will have to sort it out properly.

(I have similar worries about the way you've disabled some incompatible pointer errors, but ... no time to dig in and see if that is important at present).

Reply:
Well, given these are optional additions not needed for the compiler, we could just drop the subport.

I have opened an issue with upstream already re multiple errors, but looks like it was pretty dormant for years.